### DISCUSSION TITLE
Custom git registry to work with branches

### DISCUSSION THREAD

**Discussion by srgbtl** [microsoft/vcpkg#19704] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
As a developer, I want to be sure that a new port in the custom git registry is consumable before merging to main.
1. New branch is created in custom registry repo and the new port is added
2. In repo that uses this registry I am changing baseline to point a commit in branch and build the project
3. If the project is successfully built I am sure I can merge my changes to the main branch.

**Proposed solution**
Add a new field to [Registry object](https://github.com/microsoft/vcpkg/blob/6e80675573a63332a2625935e305f62cd7af497d/docs/users/registries.md#registry-objects) to specify a branch to use. If the field is not specified HEAD is used.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by srgbtl** [comment]
Just found #17726 and https://github.com/microsoft/vcpkg-tool/pull/99
@strega-nil-ms is there any progress on this [PR](https://github.com/microsoft/vcpkg-tool/pull/99)? I might help with this PR. I have an environment ready to test this.
CC: @BillyONeal @ras0219-msft

**Comment by strega-nil-ms** [comment]
@srgbtl unfortunately, I've been busy with other stuff right now, so I've not been able to work on it; if you'd be up for it, it'd be great if you could take up the torch.

**Comment by srgbtl** [comment]
@BillyONeal @ras0219-msft Looking forward to hear any feedback and/or vision on this issue. We really need this feature, and I'm wondering whether to switch to fork to use it.

**Comment by srgbtl** [comment]
New key was added in https://github.com/microsoft/vcpkg-tool/pull/155
Usage:
```
    {
        "default-registry": null,
        "registries": [
            {
                "kind": "git",
                "repository": "myrepourl",
                "baseline": "baseline1",
                "reference": "ref1",
                "packages": ["vcpkg-internal-e2e-test-port"]
            },
            {
                "kind": "git",
                "repository": "myrepourl",
                "baseline": "baseline2",
                "reference": "ref2",
                "packages": ["vcpkg-internal-e2e-test-port2"]
            }
        ]
    }
```
